Fix SI-3836 not-really-ambiguous import detection.

paulp opened the pull request:
Normalize types before declaring that two imports are ambiguous,
because they might be the same thing.  Review by @moors.

paulp committed:
Fix SI-3836 not-really-ambiguous import detection.

Normalize types before declaring that two imports are ambiguous,
because they might be the same thing.  Review by @moors.

paulp committed:
Small adjustment to ambiguity logic.

I think I might have ambiguified some previously unambiguous
value imports with that formulation.  This should be better.

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/431/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Failed - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/431/

paulp commented:
PLS REBUILD ALL

paulp committed:
Restructure the ambiguity check a little further.

I thought I'd arranged things so I could drop a condition,
but neg/imp2.scala told me otherwise.

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/433/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Failed - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/433/

paulp commented:
Time to reimagine this one.
